tests/engine/test_calc_job.py

```diff
@@ -74,6 +74,62 @@ def prepare_for_submission(self, folder):
         return calcinfo
 
 
+@pytest.mark.requires_rmq
+class MultiCodesCalcJob(CalcJob):
+    """`MultiCodesCalcJob` implementation to test the calcinfo with multiple codes set.
+
+    The codes are run in parallel. The codeinfo1 is set to run without mpi, and codeinfo2
+    is set to run with mpi.
```

Comment:
Mmm so I know technically all other tests with the `FileCalcJob` run """serially""", but shouldn't we also check that it works correctly when providing two codes like you do below?

You could just define an input flag in the calcjob and have that activate whether it should set  `calcinfo.codes_run_mode = CodeRunMode.PARALLEL` or not in the `prepare_for_submission`. Then you can just parametrize the test you already made.

Reply:
Sorry @ramirezfranciscof, I am not sure I fully understand your suggestion. Do you mean there is no need to have new `MultiCodeCalcJob` but I can test it in `FileCalcJob` with two codes? 

If so. Hmmm... That makes sense, I can try it.

Reply:
Mmm no, sorry I may not have been very clear.

I was just pointing out that you were testing `CodeRunMode.PARALLEL` but not `CodeRunMode.SERIAL` for multiple codes. I only mentioned `FileCalcJob` as a recognition that in principle we are implicitly using `CodeRunMode.SERIAL` in the other tests that were already there and used that prior class, but they were always testing it when there was only one code. The suggestions I made were modifications on `MultiCodeCalcJob` so that you could also use that to test `CodeRunMode.SERIAL` with multiple codes too:

```
class MultiCodeCalcJob(CalcJob):
    (...)
    spec.input('parallel_run', valid_type=Bool, default=False)
    (...)
    if parallel_run:
        calcinfo.codes_run_mode = CodeRunMode.PARALLEL
    else:
        calcinfo.codes_run_mode = CodeRunMode.SERIAL
   (...)


@pytest.mark.requires_rmq
@pytest.mark.usefixtures('clear_database_before_test')
@pytest.mark.parametrize('parallel_run', [True, False])
def test_multi_codes_run_parallel(aiida_local_code_factory, file_regression):
    (...)
    inputs = {
        parallel_run: parallel_run,
    (...)
```

Is this clearer now? Sorry for the misunderstanding; I think I personally preferred when you had the dedicated `MultiCodeCalcJob` for your multiple code test but it is ok if you don't want to change that back. The important thing for me was to try both options.